velox/functions/prestosql/types/BigintEnumType.h

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <boost/algorithm/string.hpp>
+#include <map>
+
+#include "folly/CPortability.h"
+#include "velox/type/Type.h"
+
+namespace facebook::velox {
+
+class BigintEnumType : public BigintType {
+ public:
+  explicit BigintEnumType(
+      const std::string& enumName,
+      std::map<std::string, int64_t> enumMap)
+      : BigintType(/*providesCustomComparison*/ true),
+        enumName_(enumName),
+        enumMap_(std::move(enumMap)) {}
+
+  // Creates a BigintEnumType with the given name and enum map and inserts into
+  // the static map of instances.
+  static std::shared_ptr<const BigintEnumType> create(
+      const std::string& enumName,
+      const std::map<std::string, int64_t>& enumMap);
+
+  // Returns the enum type if an enum type has been created with the given name;
+  // otherwise returns nullopt.
+  static std::optional<std::shared_ptr<const BigintEnumType>> get(
+      const std::string& enumName);
+
+  int32_t compare(const int64_t& left, const int64_t& right) const override {
+    return left < right ? -1 : left == right ? 0 : 1;
+  }
+
+  uint64_t hash(const int64_t& value) const override {
+    return folly::hasher<int64_t>()(value);
+  }
+
+  bool equivalent(const Type& other) const override {
+    return this->name() == other.name();
+  }
+
+  const char* name() const override {
+    return enumName_.c_str();
+  }
+
+  std::string toString() const override {
+    return name();
+  }
+
+  std::map<std::string, int64_t> getEnumMap() const {
```

Comment:
Use const & for the return type to avoid unnecessary copying.

Please, review coding style and update. Getters should not have "get" prefix.